### PR TITLE
Drop rebinding on Service and workload/application changes

### PR DESCRIPTION
**Pull request by Avni-Sharma**
### Motivation:
Improving the stability of the SBO prompted reducing watches which would reduce the load on the API Server.

### Changes:
Watch only SBR resources and drop watches for services and application workloads.
Remove test cases, unit and acceptance, that watches on services and workloads.

### Testing:
Run `make test-acceptance`

### CODE REVIEW

File: controllers/mapper.go [hunk 1]
```diff
@@ -206,7 +68,6 @@ type Referable interface {
 
 type K8STypeLookup interface {
 	ResourceForReferable(obj Referable) (*schema.GroupVersionResource, error)
-	KindForReferable(obj Referable) (*schema.GroupVersionKind, error)
```
Comment (pedjak) [inline controllers/mapper.go:209]: if this is not used anywhere, then please remove the implementation method from the same file.

File: controllers/mapper_test.go [hunk 2]
```diff
@@ -59,29 +58,7 @@ func TestSBRRequestMapperMap(t *testing.T) {
...
-							Kind:       "Secret",
-						},
-					},
-				}
-			},
-			expectedRequestsLen: 0,
-		},
-		{
-			description: "resource declared as application of a service binding",
+			description: "Service Binding resource declared",
```
Comment (pedjak) [inline controllers/mapper_test.go:61]: can you also say something about test expectations?

**Review by pedjak — COMMENTED** [review]
overall looks good, just a few small things to be addressed.

**Comment by pmacik** [comment]
This PR has a pretty impressive impact on performance in a loaded cluster. The memory usage dropped from  up to ~11GiB to up to ~80 MiB (~140x less memory usage) and the resulted CPU usage dropped from up to ~4.5 vCPU to up to ~0.0034 vCPU (~1323,5x less CPU usage):
image
as opposed to what the performance was without this patch:
image

**Review by pedjak — APPROVED** [review]

**Pull request merged**